app/src/main/java/fr/free/nrw/commons/upload/UploadModule.java

```diff
@@ -40,4 +45,13 @@ public abstract DepictsContract.UserActionListener bindsDepictsPresenter(
             DepictsPresenter
             presenter
     );
+
+    @Provides
+    public static UploadClient provideUploadClient(final UploadInterface uploadInterface,
+        @Named(NetworkingModule.NAMED_COMMONS_CSRF) final CsrfTokenClient csrfTokenClient,
+        final PageContentsCreator pageContentsCreator, final FileUtilsWrapper fileUtilsWrapper,
+        final Gson gson) {
+        return new UploadClient(uploadInterface, csrfTokenClient, pageContentsCreator,
+            fileUtilsWrapper, gson, System::currentTimeMillis);
```

Comment:
The _functional interface_ `TimeProvider` is designed to let us pass a lambda that returns a `Long` value e.g.
```kotlin
{ System.currentTimeMillis() }
```
Which then simplifies to the _method reference_ you see here.  All this facilitates massing a `mock<TimeProvider>()` in unit tests giving us control of the time we use in the upload client 😄 

https://kt.academy/article/ek-function-types